tools/plugins/com.liferay.ide.project.ui/src/com/liferay/ide/project/ui/modules/ext/NewModuleExtWizard.java

```diff
@@ -16,21 +16,60 @@
 
 import com.liferay.ide.core.util.CoreUtil;
 import com.liferay.ide.project.core.modules.ext.NewModuleExtOp;
+import com.liferay.ide.project.core.util.LiferayWorkspaceUtil;
 import com.liferay.ide.project.ui.ProjectUI;
 import com.liferay.ide.project.ui.modules.BaseProjectWizard;
 
 import org.eclipse.core.resources.IProject;
+import org.eclipse.jface.wizard.IWizardPage;
 import org.eclipse.sapphire.ui.def.DefinitionLoader;
+import org.eclipse.sapphire.ui.forms.swt.SapphireWizardPage;
+
+import org.osgi.framework.Version;
 
 /**
  * @author Charles Wu
+ * @author Terry Jia
  */
 public class NewModuleExtWizard extends BaseProjectWizard<NewModuleExtOp> {
 
 	public NewModuleExtWizard() {
 		super(_createDefaultOp(), DefinitionLoader.sdef(NewModuleExtWizard.class).wizard());
 	}
 
+	@Override
+	public IWizardPage[] getPages() {
+		IWizardPage[] wizardPages = super.getPages();
+
+		if (!_firstErrorMessageRemoved && (wizardPages != null)) {
+			SapphireWizardPage wizardPage = (SapphireWizardPage)wizardPages[0];
+
+			try {
+				IProject workspaceProject = LiferayWorkspaceUtil.getWorkspaceProject();
+
+				Version liferayWorkspaceVersion = new Version(
+					LiferayWorkspaceUtil.guessLiferayWorkspaceVersion(workspaceProject));
+
+				Version version70 = new Version("7.0");
+
+				if (CoreUtil.compareVersions(liferayWorkspaceVersion, version70) > 0) {
```

Comment:
This should be added to the validation service in the Op instead of in the wizard.  Can you move it there?